### DISCUSSION TITLE
Own algorithms - how to include our own algorithms

### DISCUSSION THREAD

**Discussion by fabricebrito** [developmentseed/titiler#699] [Q&A] [answered]
Hi @vincentsarago

I'm coming back at you to understand what is the best deployment strategy for including our own algorithms in titiler.

We'd like to avoid forking and build/deploy as that comes with an additional maintenance effort for aligning future features and releases.

Do you see alternative options to register "own algorithms" and start the titiler app?

**Comment by vincentsarago** [comment, marked as answer]
Hi @fabricebrito 

> We'd like to avoid forking and build/deploy

TiTiler is a full python module and should not be forked but installed and extended if needed. This way you can make sure to reduce the maintenance while this repo move forward.

The easiest way to include your own algorithm is to follow: https://developmentseed.org/titiler/advanced/Algorithms/#dependency

**Comment by fabricebrito** [comment]
Hi @vincentsarago, ok, understood.

I did a new main.py including blocks like this one:

```python
stac = MultiBaseTilerFactory(
        reader=STACReader,
        router_prefix="/stac",
        extensions=[
            stacViewerExtension(),
        ],
        process_dependency=PostProcessParams,
    )
``` 

I'm happy with the outcome, thank you for taking the time 

Fabrice
